src/lib/mongoBackend/safeMongo.h

```diff
@@ -32,47 +32,57 @@
 
 #include "mongo/client/dbclient.h"
 
+/* Some macros to make the usage of these function prettier */
+#define getObjectFieldF(b, field)           getObjectField(b, field, __FUNCTION__, __LINE__)
+#define getStringFieldF(b, field)           getStringField(b, field, __FUNCTION__, __LINE__)
+#define getIntFieldF(b, field)              getIntField(b, field, __FUNCTION__, __LINE__)
+#define getLongFieldF(b, field)             getLongField(b, field, __FUNCTION__, __LINE__)
+#define getIntOrLongFieldAsLongF(b, field)  getIntOrLongFieldAsLong(b, field, __FUNCTION__, __LINE__)
+#define getBoolFieldF(b, field)             getBoolField(b, field, __FUNCTION__, __LINE__)
+#define getFieldF(b, field)                 getField(b, field, __FUNCTION__,  __LINE__)
+#define nextSafeOrErrorF(c, r, err)         nextSafeOrError(c, r, err, __FUNCTION__,  __LINE__)
+
 /* ****************************************************************************
 *
 * getObjectField -
 */
-extern mongo::BSONObj getObjectField(const mongo::BSONObj& b, const std::string& field);
+extern mongo::BSONObj getObjectField(const mongo::BSONObj& b, const std::string& field, const std::string& caller = "<none>", int line = 0);
```

Comment:
Nah, why bother.
Leave it the way it is.

NTC.

Only 'interesting' thing to fix this very slight problem would be creating a macro with the same name as the function. A macro accepting two params and then calling the function with four. Something like:

    #define getObjectField(a, b)   getObjectField(a, b, __FUNCTION__, __LINE)

However, this is pretty cryptic code.
Better too keep it with default params.